src/index.ts

```diff
@@ -96,8 +99,16 @@ export class WaxJS {
     ) => void;
     metricURL?: string;
     returnTempAccounts?: boolean;
+    chainName?: string;
   }) {
     this.rpc = new JsonRpc(rpcEndpoint);
+
+    this.chainName = chainName || null;
+    this.chainId = null;
+    if (chainName) {
```

Comment:
we can remove the if here. I think it is cleaner to not check `if(chainName)` just do that logic in the `setupChain` function, so we will always call `await setupChain(chainName)`

Reply:
updated base on comment 👍